### PR TITLE
Make musl test a container job

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
The purpose of this PR is to speed up our CI by speeding up the slowest step. Rather than running docker as part of this process, we just run this entire job in the alpine container. I'm hoping this will recover a significant amount of performance.

# Review

- [ ] Add a short description of the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Review by syrusakbary — APPROVED** [review]

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -45,28 +45,40 @@ jobs:
...
             cross_compilation_artifact_name: 'cross_compiled_from_linux'
             run_integration_tests: true
+            use_sccache: true
           - build: macos-x64
             os: macos-latest
             llvm_url: 'https://github.com/wasmerio/llvm-custom-builds/releases/download/11.x/darwin-amd64.tar.gz'
             artifact_name: 'wasmer-darwin-amd64'
             cross_compilation_artifact_name: 'cross_compiled_from_mac'
             run_integration_tests: true
+            use_sccache: false
```
Comment (syrusakbary) [inline .github/workflows/main.yaml:55]: We probably want this to be true!

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -45,28 +45,40 @@ jobs:
...
           - build: macos-arm64
             os: macos-11.0
             target: aarch64-apple-darwin
             artifact_name: 'wasmer-darwin-arm64'
+            use_sccache: false
```
Comment (syrusakbary) [inline .github/workflows/main.yaml:60]: We probably want this to be true!

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -45,28 +45,40 @@ jobs:
...
           - build: windows-x64
             os: windows-latest
             # llvm_url: 'https://github.com/wasmerio/llvm-custom-builds/releases/download/11.x/windows-amd64.tar.gz'
             artifact_name: 'wasmer-windows-amd64'
             cross_compilation_artifact_name: 'cross_compiled_from_win'
             run_integration_tests: true
+            use_sccache: false
```
Comment (syrusakbary) [inline .github/workflows/main.yaml:67]: We probably want this to be true!

Comment (MarkMcCaskey) [reply .github/workflows/main.yaml:67]: I suspect that sccache isn't doing anything on most of our platforms, just testing it out now

Comment (syrusakbary) [reply .github/workflows/main.yaml:67]: Windows is the only one that I'm more unsure. On the rest I think it should shave significant time.

Comment (MarkMcCaskey) [reply .github/workflows/main.yaml:67]: Alright, you're right! Sccache definitely makes them all faster. Glad I checked though!

**Pull request merged**